rust/tools/authorization-logic/src/souffle/souffle_emitter.rs

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2021 The Raksha Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::{ast::*, souffle::datalog_ir::*};
+use std::collections::HashSet;
+
+pub struct SouffleEmitter {
+    // This is a set of predicates that need to be declared. This set is
+    // populated by side effect while traversing the program body.
+    decls: HashSet<AstPredicate>,
+}
+
+impl<'a> SouffleEmitter {
```

Comment:
(nit) I don't think the lifetime needs to be here, it may be just on the methods that need it, and I'm not even sure any method actually needs it anyways.

Reply:
You're right and now they're all gone.